### ISSUE TITLE
getPrivateKey - Failed to retrieve private key from keystore

### ISSUE THREAD

**Issue by guillaumep-MF** [networknt/light-oauth2#99]
I'm not sure what happened with my setup, but from the light-oauth2.git project I can't emit any tokens anymore. The primary.jks/secondary.jks files are under the db/config I use, everything seems to be in place except I just get an NPE.

Maybe add a little more information to why the Private key can't be read is in order?

**Comment by stevehu** [comment]
@guillaumep-MF One of the customers requires that we remove all the default config files from the package for security reasons. I think it is a good idea to pass the configuration files to the container so that nobody accidentally uses any default config which is a security problem. I think you are trying to start the services with the docker-compose files in light-oauth2 which is for light-oauth2 developers. If you want to use it, I would recommend using the docker-compose from light-docker which contains all the default config files. The reason private key cannot be found might be the wrong location in jwt.yml file.

**Comment by guillaumep-MF** [comment]
@stevehu No doubt I have some issue with my configuration or Docker itself. The ticket is more about having the token service behave in an eager manner with files that are critical to its functioning state. Maybe you can have this service without access to a keystore, but when it's supposed to use it and its critical I would have it check the existence of the file before you ever get to issue tokens.

It seems more reasonable, than do nothing and then wait until someone asks for a token and just log an error with a Null message.

**Comment by stevehu** [comment]
I agree that we need to handle the error more graceful. Do you recommend checking the keystore during the startup? Let's keep this issue open until we have a better solution implemented.

**Comment by guillaumep-MF** [comment]
Yes, I think any resource that is critical to the functioning of a service (micro or not), should be eagerly initiated. You're just closer to the problems raised if any and it's easier to elect the right strategy then.
